lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -86,17 +89,26 @@ def set_host_and_port_and_method_info_on_desc
             end
           end
 
-          def server_options(headers)
+          def grpc_headers(metadata)
+            metadata.reject { |k, v| DT_KEYS.include?(k) }
+          end
+
+          def grpc_params(metadata, streamer_type, host, port, method)
+            {headers: grpc_headers(metadata),
+             uri: "grpc://#{host}:#{port}/#{method}",
+             method: method,
+             type: streamer_type}
+          end
+
+          def trace_options(metadata, streamer_type)
             host = instance_variable_get(INSTANCE_VAR_HOST)
             port = instance_variable_get(INSTANCE_VAR_PORT)
             method = instance_variable_get(INSTANCE_VAR_METHOD)
-            {
-              request: {
-                headers: headers,
-                uri: "grpc://#{host}:#{port}/#{method}",
-                method: method
-              }
-            }
+            # TODO: gRPC - should category be 'gRPC'
+            # TODO: gRPC - should transaction name start with 'External/'?
```

Comment:
I don't think the name should start with "External/" because the instrumented application is generating the transaction. Sidekiq uses "OtherTransaction/SidekiqJob", so we could follow in that pattern and do something like "OtherTransaction/gRPC". I don't know if that would change the category of the transaction to non-web.